packages/wonder-blocks-tooltip/util/is-obscured.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+/**
+ * Determine if an element is obscured by other elements.
+ *
+ * This uses document.elementFromPoint to see if the given element is being
+ * overdrawn by another element. Note that this won't work if the given element
+ * has `pointer-events: none`.
+ */
+export default function isObscured(element: Element) {
+    // TODO(somewhatabstract): We should be smarter in this algorithm and
+    // actually look at the intersection of the elements doing the obscuring
+    // just as we already do with our scroll parent intersections. That way we
+    // can not only check that the entire element is obscured, but think about
+    // partial obscurement so we can move the tooltip bubble when it's anchor
+    // point is not visible.
+
+    // Before we assume we're visible let's check to see if something else
+    // is obscuring us.  Here we make the assumption that if our topleft and
+    // bottomright corners are covered by something, we're not visible.
+    // There are ways that this can still not work, such as different
+    // elements only covering those points and the remainder being visible,
+    // or if some covering element has none for pointer-events style, but
+    // those edge cases shouldn't bother the main usages for this method.
+
+    // NOTE: If the anchor element has `pointer-events: none`, we're always
+    // going to end up hiding, so, you know, probably don't do that.
```

Comment:
I don't see any code that checks for `pointer-events: none`.  Could you explain more how this works?

Reply:
`elementFromPoint` doesn't return things that have `pointer-events: none`. We could add a runtime check for that on our anchors but that seems a little heavy-handed for what amounts to a cornercase.

Reply:
I have updated the commentary for this.